Suggested edits for identity doc updates (#29339)
hashicorp/vault: website/content/docs/upgrading/deduplication/entity-group.mdx

[WARN]  identity: entity "entity-case" with namespace ID "root" duplicates 2 others: id=290a643d-6043-da5e-943f-3a3d09e4ecbd force_deduplication="would rename to entity-case-290a643d-6043-da5e-943f-3a3d09e4ecbd"
[WARN]  identity: entity "entity-case" with namespace ID "root" duplicates 2 others: id=b0141be5-3f03-a1c7-a57b-02f045d04426 force_deduplication="would not rename"
[WARN]  identity: entity "entity-cAsE" with namespace ID "sYMXY" duplicates 2 others: id=95f0743b-a1d5-26da-b4ef-a50490da0787 force_deduplication="would rename to entity-cAsE-95f0743b-a1d5-26da-b4ef-a50490da0787"
[WARN]  identity: entity "entity-case" with namespace ID "sYMXY" duplicates 2 others: id=9d3be96f-490a-9625-3118-896bd2b3a5f3 force_deduplication="would rename to entity-case-9d3be96f-490a-9625-3118-896bd2b3a5f3"
[WARN]  identity: entity "entity-case" with namespace ID "sYMXY" duplicates 2 others: id=d82231d2-3716-6b9c-e80d-7d09c0409739 force_deduplication="would not rename"
[WARN]  identity: end of entity duplicates

> [WARN]  identity: 2 group duplicates found

Reviewer: ```suggestion
```
Line is duplicated

[WARN]  identity: 2 group duplicates found
[WARN]  identity: group "group-case" with namespace ID "root" duplicates 2 others: id=8ad26e0c-8cf6-5b67-7c77-6571fa374f34 force_deduplication="would rename to group-case-8ad26e0c-8cf6-5b67-7c77-6571fa374f34"
[WARN]  identity: group "group-cAsE" with namespace ID "root" duplicates 2 others: id=9fe86ea0-f80c-1131-5be1-1d6e3b70237f force_deduplication="would rename to group-cAsE-9fe86ea0-f80c-1131-5be1-1d6e3b70237f"

> [WARN]  identity: group "group-case" with namespace ID "root" duplicates 2 others: id=32dd070c-c1f8-c796-9a71-15887014b813 force_deduplication="would not rename"

Reviewer: Technically we changed which of these say "would not rename" - it would now be the first one not the last. But I don't think that matters. for the docs here?

Author: It's always better if the examples in the dogs match the actual output. I'll fix it.

[WARN]  identity: group "group-case" with namespace ID "sYMXY" duplicates 2 others: id=f11277b3-d985-4d72-d2e9-9c8c6c0db02c force_deduplication="would rename to group-case-f11277b3-d985-4d72-d2e9-9c8c6c0db02c"
[WARN]  identity: group "group-cAsE" with namespace ID "sYMXY" duplicates 2 others: id=7c753d07-b0d9-e13b-6184-48247b8f7504 force_deduplication="would not rename"
[WARN]  identity: end of group duplicates
```

Entity and group duplicates are reported as in the examples above. The
resolution steps are the same for both.

Entities or groups may exist with duplicate names. These might be exact matches,
or may differ in case such as `Admin` and `admin`. In either case, the
resolution is the same.

Note that names are only duplicates if they are in the same namespace.

Possible unexpected behavior caused by these types of duplicates includes:
 * Only one of them will be returned when looking up or listing by name
 * Before Vault 1.19.0, which one is returned might vary depending on which

>    server is hit or even after a seal and unseal on the same server

Reviewer: ```suggestion
   server handles the request or even after a seal and unseal on the same server
```

Fix duplicate identities for entities and groups targeted for renaming.

<Tip title="Assumptions">

- You are running Vault 1.19 or later.
- You have [deduplication **renaming** targets in your system logs](/vault/docs/upgrading/identity-deduplication).
- You have admin permission on the relevant Vault server or cluster.

</Tip>

## Why duplicates happen

Historical bugs in Vault allowed authN mounts to create duplicate entries of the
same entity or group name if the names used difference cases. By default, Vault
uses case-insensitive matching for names and considers `bob` and `BOB` to be the
same name. But, past bugs have allowed for identical or different-case
duplicates in entity and group names.

Duplicate entity and group names can cause unexpected behavior and make working
with Vault more difficult:

- Fetching entities/groups by name returns the first identity found. Depending
  on your Vault version, the identity returned can vary depending on the server
  handling the request and when the last seal/unseal event occurred.
- Deleting entities/groups by name is not guaranteed to delete all duplicates.
- All lookup requests **must** use ID to find the right identity.

> - Listing entities/groups by ID returns all duplicates instead of a specific,

Reviewer: ```suggestion
```
This is actually expected behavior. I originally included it to highlight the inconsistency between different list methods but it was confusing to other reviewers already so I suggest we just remove it.

2025-01-28T13:15:13.641-0800 [WARN]  identity: 2 group duplicates found
2025-01-28T13:15:13.641-0800 [WARN]  identity: group "group-case" with namespace ID "root" duplicates 2 others: id=8ad26e0c-8cf6-5b67-7c77-6571fa374f34 force_deduplication="would not rename"
2025-01-28T13:15:13.641-0800 [WARN]  identity: group "group-cAsE" with namespace ID "root" duplicates 2 others: id=9fe86ea0-f80c-1131-5be1-1d6e3b70237f force_deduplication="would rename to group-cAsE-9fe86ea0-f80c-1131-5be1-1d6e3b70237f"
2025-01-28T13:15:13.641-0800 [WARN]  identity: group "group-case" with namespace ID "root" duplicates 2 others: id=32dd070c-c1f8-c796-9a71-15887014b813 force_deduplication="would rename to group-case-32dd070c-c1f8-c796-9a71-15887014b813"
2025-01-28T13:15:13.641-0800 [WARN]  identity: group "group-case" with namespace ID "sYMXY" duplicates 2 others: id=8aaeff7e-7343-c883-1e0c-c5c9968f75a5 force_deduplication="would not rename"
2025-01-28T13:15:13.641-0800 [WARN]  identity: group "group-case" with namespace ID "sYMXY" duplicates 2 others: id=f11277b3-d985-4d72-d2e9-9c8c6c0db02c force_deduplication="would rename to group-case-f11277b3-d985-4d72-d2e9-9c8c6c0db02c"

> 2025-01-28T13:15:13.641-0800 [WARN]  identity: group "group-cAsE" with namespace ID "sYMXY" duplicates 2 others: id=7c753d07-b0d9-e13b-6184-48247b8f7504 force_deduplication="would rename to group-cAsE-07c753d07-b0d9-e13b-6184-48247b8f7504"

Reviewer: Val made a great point on the other PR that `canonical_id` might not be obvious unless people are more familiar with our API. It's used to remain consistent with the API, See https://developer.hashicorp.com/vault/api-docs/secret/identity/entity-alias#canonical_id. That's the name of the field that holds the specific entity ID the alias is linked to.

But it might be a nice call to just mention that explicitly in the description below? I defer to your judgement though Sarah!

Author: I think mentioning canonical ID makes sense, but not in this context since it's about entity/group renaming and the example logs don't include a `canonical_id` field. I'll add it to the same-case guide :)
